src/pages/OfferNew/Perils/PerilModal.tsx

```diff
@@ -0,0 +1,262 @@
+  display: flex;
+  flex-flow: row;
+  margin: 0 -12px;
+  overflow: hidden;
+  position: relative;
+  box-sizing: border-box;
+`
+
+const PickerItem = styled('button')`
+  width: 100px;
+  flex-shrink: 0;
+  padding: 8px 8px 10px 8px;
+  margin: 0 8px;
+  background: none;
+  border: none;
+  display: flex;
+  flex-flow: column;
+  justify-content: space-between;
+  align-items: center;
+  cursor: pointer;
+
+  svg {
+    width: 44px;
+    height: 44px;
+  }
+
+  :focus {
+    outline: none;
+  }
+`
+
+const PickerItemLabel = styled('div')`
+  font-size: 15px;
+  letter-spacing: -0.23px;
+  text-align: center;
+  white-space: nowrap;
+  color: ${colorsV2.darkgray};
+`
+
+interface PerilItemsContainerProps {
+  currentPeril: number
+  transition: boolean
+}
+
+const PerilItemsContainer = styled('div')<PerilItemsContainerProps>`
+  position: relative;
+  height: 88px;
+  display: flex;
+  ${(props) =>
+    props.transition && `transition: all ${TRANSITION_MS}ms ease-in-out;`}
+
+  ${(props) =>
+    `transform: translateX(${
+      props.currentPeril !== 0
+        ? `calc(-33.3333333333333% - ${(props.currentPeril - 12 - 1) *
+            (100 + 16) +
+            8}px)`
+        : `calc(-33.3333333333333% + 108px)`
+    });`}
+`
+
+const Indicator = styled('div')`
+  position: absolute;
+  width: 100px;
+  height: 2px;
+  bottom: 0;
+  left: 0;
+  background-color: ${colorsV2.black};
+  border-top-left-radius: 2px;
+  border-top-right-radius: 2px;
+  transition: transform 0.15s ease-in-out;
+  left: 116px;
+`
+
+const Gradient = styled('div')`
+  height: 80px;
+  width: 100px;
+  position: absolute;
+  display: flex;
+  align-items: center;
+  bottom: 8px;
+`
+
+const LeftGradient = styled(Gradient)`
+  left: 0;
+  background: linear-gradient(
+    to right,
+    ${colorsV2.white} 30%,
+    ${hexToRgba(colorsV2.white, 0)} 100%
+  );
+`
+
+const RightGradient = styled(Gradient)`
+  right: 0;
+  justify-content: flex-end;
+  background: linear-gradient(
+    to left,
+    ${colorsV2.white} 30%,
+    ${hexToRgba(colorsV2.white, 0)} 100%
+  );
+`
+
+const DirectionButton = styled('button')`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  cursor: pointer;
+  background: none;
+  border: none;
+  svg {
+    * {
+      transition: all 0.15s ease-in-out;
+      fill: ${colorsV2.gray};
+    }
+  }
+  :focus {
+    outline: none;
+  }
+  &:hover {
+    svg {
+      * {
+        fill: ${colorsV2.violet500};
+      }
+    }
+  }
+`
+
+const BackButton = styled(DirectionButton)`
+  margin-left: 10px;
+  justify-content: flex-start;
+`
+
+const ForwardButton = styled(DirectionButton)`
+  margin-right: 10px;
+  justify-content: flex-end;
+`
+
+const Content = styled('div')`
+  padding: 24px;
+`
+
+const createPerilItems = (
+  perils: Peril[],
+  setCurrentPeril: (index: number) => void,
+  actionsAllowed: boolean,
+) => {
+  const items = []
+  for (let i = 0; i < perils.length * 3; i++) {
+    const index = i % perils.length
+    items.push(
```

Comment:
i don't like loops 😬 did we at least _try_ to do this with map/reduce?

Reply:
Managed to fix it now with a map, I don't like them either but I think it's okay in some occasions when it makes the logic easier to understand!